app/controllers/application_controller.rb

```diff
@@ -5,6 +5,132 @@ class ApplicationController < Sinatra::Base
   configure do
     set :public_folder, 'public'
     set :views, 'app/views'
+    enable :sessions
+    set :session_secret, "password_security"
   end
 
-end
\ No newline at end of file
+  get '/' do
+    if logged_in?
+      redirect :'/tweets'
+    else
+      erb :index
+    end
+  end
+
+  get '/signup' do
+    if !logged_in?
+      erb :'/users/create_user'
+    else
+      redirect '/tweets'
+    end
+  end
+
+  post '/signup' do
+    if !params[:username].empty? && !params[:email].empty? && !params[:password].empty?
+      @user = User.create(params)
+      session[:user_id] = @user.id
+      redirect :'/tweets'
+    else
+      redirect :'/signup'
+    end
+  end
+
+  get '/login' do
+    if !logged_in?
+      erb :'/users/login'
+    else
+      redirect :'/tweets'
+    end
+  end
+
+  post '/login' do
+    @user = User.find_by(username: params[:username])
+    session[:user_id] = @user.id
+    redirect :'/tweets'
+  end
+
+  get '/users/:id' do
+    @user = User.find(params[:id])
+    erb :'/users/show'
+  end
+
+  get '/tweets' do
+    if logged_in?
+      @user = current_user
+      erb :'/tweets/tweets'
+    else
+      redirect :'/login'
+    end
+  end
+
+  get '/tweets/new' do
+    if logged_in?
+      erb :'/tweets/create_tweet'
+    else
+      redirect '/login'
+    end
+  end
+
+  post '/tweets' do
+    if !params[:content].empty?
+      @tweet = Tweet.create(params)
+      @tweet.update(user_id: current_user.id)
```

Comment:
The tweet creation and user association can both be accomplished in the same step:

``` ruby
@tweet = Tweet.create(content: params[:content], user: current_user)
```